packages/repack/android/src/main/java/com/callstack/repack/ChunkManagerPackage.kt

```diff
@@ -1,17 +1,30 @@
 package com.callstack.repack
 
-import com.facebook.react.ReactPackage
-import com.facebook.react.bridge.NativeModule
-import com.facebook.react.bridge.ReactApplicationContext
-import com.facebook.react.uimanager.ViewManager
+import com.facebook.react.bridge.*
+import com.facebook.react.TurboReactPackage
+import com.facebook.react.module.model.ReactModuleInfo
+import com.facebook.react.module.model.ReactModuleInfoProvider
 
+class ScriptManagerPackage : TurboReactPackage() {
+    override fun getModule(name: String?, reactContext: ReactApplicationContext): NativeModule? =
+            if (name == ScriptManagerModuleImpl.NAME) {
+                ScriptManagerModule(reactContext)
+            } else {
+                null
+            }
 
-class ScriptManagerPackage : ReactPackage {
-    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
-        return listOf(ScriptManagerModule(reactContext))
-    }
-
-    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
-        return emptyList()
+    override fun getReactModuleInfoProvider() = ReactModuleInfoProvider {
+        val isTurboModule: Boolean = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED
+        mapOf(
+                ScriptManagerModuleImpl.NAME to ReactModuleInfo(
+                        ScriptManagerModuleImpl.NAME, // name
+                        ScriptManagerModuleImpl.NAME, // className
+                        false, // canOverrideExistingModule
+                        false, // needsEagerInit
```

Comment:
I don't know what exactly `needsEagerInit` means, but I'd say Repack should be initialized & available as soon as possible, especially in the Module Federation architecture we're now treating as a recommended one.

Take a look at the Super App Showcase Host App – https://github.com/callstack/super-app-showcase/blob/main/packages/host/index.js ScriptManager is used before the App component is registered.

Of course, what I'm saying here might be completely wrong since I'm not that familiar with the New Arch – if that's the case, sorry for the confusion :)

Reply:
On repack side, i'm not sure how eager it needs to be initialised (even though i feel it's still initialised and available), however this is the default spec from docs and i'm sure it can be set to Eagerly initialise.